src/components/Select/Select.tsx

```diff
@@ -0,0 +1,70 @@
+"use client";
+import { useEffect, useState } from "react";
+import { Select as MuiSelect, SelectChangeEvent } from "@mui/material";
+import { MenuItem } from "@mui/material";
+import styled from "@emotion/styled";
+import type { SelectProps } from "@mui/material";
+
+export type StyleSelectType = {
+  height?: number;
+  width?: number;
+};
+
+interface ISelect extends StyleSelectType, SelectProps {
+  items: Array<string>;
+  placeholder?: string;
+  currentSelectedData?: string;
+  onChangeValue: (value: string) => void;
+}
+
+const Select = (props: ISelect) => {
+  const { items, placeholder, height, width, currentSelectedData, onChangeValue, ...others } = props;
+  const [selected, setSelected] = useState(currentSelectedData);
+
+  const handleChange = (event: SelectChangeEvent) => {
+    const newValue: string = event.target.value;
+    setSelected(newValue);
+    onChangeValue(newValue);
+  };
+
+  useEffect(() => {
+    setSelected(currentSelectedData);
+  }, [currentSelectedData]);
+
+  return (
+    <>
+      <StyledSelect
+        value={selected}
+        onChange={handleChange}
+        displayEmpty
+        renderValue={() => {
+          if (!selected) {
+            return <em>{placeholder ? placeholder : "None"}</em>;
+          }
+
+          return selected;
+        }}
+        width={width}
+        height={height}
+        {...others}
+      >
+        {items?.map((item: string, idx: number) => (
+          <MenuItem value={item} key={`${item}-${idx}`}>
+            {item}
+          </MenuItem>
+        ))}
```

Comment:
items type이 꼭 들어오는 값이라고 나오는데 ? 이 인자가 필요 없을 것 같아용
```suggestion
        {items.map((item: string, idx: number) => (
          <MenuItem value={item} key={`${item}-${idx}`}>
            {item}
          </MenuItem>
        ))}
```

Reply:
해당 인자를 삭제하면 스토리북 실행 시  map undefined 관련된 에러가 발생하여 일단 select 컴포넌트에서는 ? 인자를 제거하지 않았어용! 해당 오류 사항은 좀 더 찾아보고 수정하려고 합니다!!

Reply:
@sians5544 스토리북에 들어가는 mockData가 문제인 것 같아오... 스토리북은 그냥 실제로 들어가는 데이터 기반으로 뿌려주기 때문에 영향은 없을거라서 ㅠㅠㅠㅠㅠ